apps/portal/src/content/docs/components/form.mdx

```diff
@@ -7,63 +7,86 @@ The `Form` component is a container for form elements, providing a structured la
 
 import { DocsPage } from "@/components/docs-page";
 
-```tsx
-import { useForm } from "react-hook-form";
+<DocsPage.ComponentExample
+  client:only
+  code={`() => {
+  const [submittedValue, setSubmittedValue] = React.useState<FormValues | null>(null);
 
-import { zodResolver } from "@hookform/resolvers/zod";
-import { z } from "zod";
+const inputRef = React.useRef<HTMLInputElement | null>(null);
 
-//...
-
-// Define the validation schema with zod
 const formSchema = z.object({
-  inputField: z.string().min(3, "Input must be at least 3 characters"),
+name: z.string().trim().min(3, "Input must be at least 3 characters"),
+
+    // 🚨 z.coerce.number() - It is mandatory for number fields
```

Comment:
It is necessary for a `number` input component. We need to explicitly highlight in the docs.

Reply:
Changes from `number` component will be returned as string always, in order to use it as number, we need to `coerce` via zod. 

https://github.com/shadcn-ui/ui/issues/421#issuecomment-1561080201